src/game-engine/state/reducers/game.reducer.js

```diff
@@ -105,6 +96,11 @@ export function game(state = initialState, action) {
 				inGame: false
 			}
 		case gameConstants.LEAVE_GAME_REQUEST:
+			return {
+				...state,
+				locked: true
+			}
+		case gameConstants.LEAVE_GAME_SUCCESS:
```

Comment:
Shouldn't lock be false if leave game is success? And we should also remove all the store data right